### PR TITLE
Make local-up-cluster.sh use /tmp for dns-related files

### PR DESCRIPTION
**Pull request by pmorie**
local-up-cluster.sh leaks files into the local directory; use /tmp instead.

@ncdc @eparis @derekwaynecarr shot out of a cannon

---

This change is [Reviewable](http://reviewable.k8s.io/reviews/kubernetes/kubernetes/22131)

### CODE REVIEW

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -378,9 +378,9 @@ function start_kubedns {
 
     if [[ "${ENABLE_CLUSTER_DNS}" = true ]]; then
         echo "Creating kube-system namespace"
-        sed -e "s/{{ pillar\['dns_replicas'\] }}/${DNS_REPLICAS}/g;s/{{ pillar\['dns_domain'\] }}/${DNS_DOMAIN}/g;" "${KUBE_ROOT}/cluster/addons/dns/skydns-rc.yaml.in" >| skydns-rc.yaml
-        sed -e "s/{{ pillar\['dns_server'\] }}/${DNS_SERVER_IP}/g" "${KUBE_ROOT}/cluster/addons/dns/skydns-svc.yaml.in" >| skydns-svc.yaml
-        cat <<EOF >namespace.yaml
+        sed -e "s/{{ pillar\['dns_replicas'\] }}/${DNS_REPLICAS}/g;s/{{ pillar\['dns_domain'\] }}/${DNS_DOMAIN}/g;" "${KUBE_ROOT}/cluster/addons/dns/skydns-rc.yaml.in" >| /tmp/skydns-rc.yaml
```
Comment (luxas) [inline hack/local-up-cluster.sh]: While we're doing this, could you 
```bash
TEMP_DIR=$(mktemp -d)
```

and use that instead of `/tmp`?
Just a question.

Comment (pmorie) [inline hack/local-up-cluster.sh]: @luxas I'd prefer that the person who added this functionality take care of the leaks -- see #22180 -- right now I'm just trying to get this script to stop littering up my working directory.

That said, I'm happy to change it if you feel strongly, let me know

**Comment by eparis** [comment]
I'm with @luxas all you did we move the problem from one place to another and didn't actually fix it.

`kube::util::ensure-temp-dir()` I think should be used here...

File: hack/local-up-cluster.sh (hunk 1 above)
Comment (luxas) [inline hack/local-up-cluster.sh]: Can't say I feel strongly, but it was just a suggestion how to do it even better, and would be nice to have.

**Comment by zmerlynn** [comment]
I'm actually with the others. Sorry for being a slowpoke to reply.

**Comment by zmerlynn** [comment]
Is this PR headed somewhere? Please reopen if so. :)

**Pull request closed without merging**